### ISSUE TITLE
Problem in CREATE_HOOK function

### ISSUE THREAD

**Issue by mahla-chr** [codedust/go-tox#9]
hi...
when I try to run examples in ubuntu, these errors occur repeatedly for each rows of hook.c (rows 27-41):

 github.com/codedust/go-tox
In file included from /usr/lib/go-1.8/src/github.com/codedust/go-tox/callbacks.go:5:0:
./hooks.c: In function 'set_callback_self_connection_status':
./hooks.c:7:3: error: too few arguments to function 'tox_callback_self_connection_status'
   tox_##x(tox, hook_##x); \
   ^

./hooks.c:27:1: note: in expansion of macro 'CREATE_HOOK'
 CREATE_HOOK(callback_self_connection_status)
 ^
In file included from /usr/lib/go-1.8/src/github.com/codedust/go-tox/callbacks.go:4:0:
/usr/local/include/tox/tox.h:779:6: note: declared here
 void tox_callback_self_connection_status(Tox *tox, tox_self_connection_status_cb *callback, void *user_data);
      ^
![image](https://user-images.githubusercontent.com/30427466/29240536-b742097c-7f7c-11e7-9531-36b1270717d3.png)

I think there is something wrong with this part of code(hook.c: line 5-7):

![image](https://user-images.githubusercontent.com/30427466/29240460-207cf8ae-7f7b-11e7-8527-a7ff18bfbda7.png)

**Comment by Omid-r** [comment]
Hi guys
I get this error too when run example.go
please guide ..

**Comment by codedust** [comment]
Thanks for reporting the issue. I could not reproduce the issue with toxcore version `0.1.9`. Which version of toxcore did you use?

**Comment by mahla-chr** [comment]
Thank you so much for your reply. 
I think the version of my toxcore is 0.1.9 too. Actually I've installed it in the last couple of weeks!
Do you have any idea about solving the problem? I tried in different computers but the errors were the same.

**Comment by codedust** [comment]
Hi @Mahla-ch,

here's what I just found out: When having a closer look at your debug log, this part shows that your go instance is not using the latest version of `c-toxcore`:

```c
In file included from /usr/lib/go-1.8/src/github.com/codedust/go-tox/callbacks.go:4:0:
/usr/local/include/tox/tox.h:779:6: note: declared here
void tox_callback_self_connection_status(Tox *tox, tox_self_connection_status_cb *callback, void *user_data);
```

When you compare this line to the latest version of `c-toxcore`, you'll notice that the two lines differ.
https://github.com/TokTok/c-toxcore/blob/master/toxcore/tox.h#L979

Could you try to uninstall all versions of toxcore on your system and (after verifying that `go-tox` has trouble including `tox.h`) install the latest version of `c-toxcore` again?
